### PR TITLE
fix: when the token can not be exchanged put the error message in condition

### PR DESCRIPTION
**Pull request by secondsun**
In the current version of the Operator if the token exchange could not be performed the failure would be logged in a condition, but the error message wasn't included. This fix updates the error handling so that the error message gets attached to the condition correctly.

See
image
vs
image

### CODE REVIEW

File: source/rhoas/src/main/java/com/openshift/cloud/controllers/ConditionUtil.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ public static String getStandarizedErrorMessage(int code, Exception e) {
         TypeReference<HashMap<String, Object>> typeRef = new TypeReference<>() {};
         errorObject = new ObjectMapper().readValue(e.getMessage(), typeRef);
         reason = errorObject.getOrDefault("reason", "").toString();
+
+        if (reason.isBlank()) {
+          reason = errorObject.getOrDefault("error_description", "").toString();
```
Comment (wtrocki) [inline source/rhoas/src/main/java/com/openshift/cloud/controllers/ConditionUtil.java:229]: This is generic method for handling conditions but here we using field that is probably specific for some API request. 
Why here then and not in the `AccessTokenSecretTool` file. Is this like futureproofing things?

Comment (secondsun) [reply source/rhoas/src/main/java/com/openshift/cloud/controllers/ConditionUtil.java:229]: No, this was a very quick hack. Looking more closely at it we probably shouldn't be using kas ApiException for an auth exception. I'll refactor today.

**Comment by wtrocki** [comment]
@secondsun How do you simulate failing token exchange for  verification?

File: source/rhoas/src/main/java/com/openshift/cloud/beans/AccessTokenSecretTool.java [hunk 2]
```diff
@@ -127,11 +130,10 @@ private String exchangeToken(String offlineToken) throws ConditionAwareException
         var json = new JsonObject(tokens);
         return json.getString("access_token");
       } else {
-        LOG.log(Level.SEVERE,
-            String.format("Exchange token failed with error %s", response.body()));
-        // Reusing API error but only with status code
-        var apiError = ConditionUtil.getStandarizedErrorMessage(response.statusCode(),
-            new ApiException(response.statusCode(), null));
+        var body = response.body();
+        var apiError = String.format("Exchange token failed with error %s", body);
```
Comment (wtrocki) [inline source/rhoas/src/main/java/com/openshift/cloud/beans/AccessTokenSecretTool.java:134]: Small risk that body would contain token information?

**Review by wtrocki — APPROVED** [review]

**Pull request merged**